### PR TITLE
Issues/commercial/1952 extra states bugfix

### PR DESCRIPTION
**Pull request by struan**
[skip changelog]

### CODE REVIEW

File: templates/web/base/report/_update_state.html [hunk 1]
```diff
@@ -1,5 +1,5 @@
 [% # Small chance of duplicates in the case of fixed - user followed by fixed - council %]
-[% SET update_state = update.problem_state %]
+[% SET update_state = update.problem_state_processed %]
 [% IF ( update_state AND update_state != global.last_state AND NOT (global.last_state == "" AND update.problem_state == 'confirmed') )
```
Comment (dracos) [inline templates/web/base/report/_update_state.html:3]: I guess for consistently end of this line should use problem_state_processed as well?

File: t/cobrand/northamptonshire.t [hunk 2]
```diff
@@ -246,6 +246,41 @@ subtest 'check further investigation state' => sub {
...
+        cobrand => 'default',
+    } );
+    $comment2->set_extra_metadata('external_status_code' => 'further');
+    $comment2->update;
+
+    FixMyStreet::override_config {
+        ALLOWED_COBRANDS => [ { northamptonshire => '.' } ],
+        MAPIT_URL => 'http://mapit.uk/',
+        COBRAND_FEATURES => {
+            extra_state_mapping => {
```
Comment (dracos) [inline t/cobrand/northamptonshire.t:270]: I don't understand what this is for.

File: t/cobrand/northamptonshire.t [hunk 3]
```diff
@@ -246,6 +246,41 @@ subtest 'check further investigation state' => sub {
...
+        confirmed => DateTime->now,
+        state => 'confirmed',
+        problem_state => 'investigating',
+        cobrand => 'default',
+    } );
+    $comment2->set_extra_metadata('external_status_code' => 'further');
+    $comment2->update;
+
+    FixMyStreet::override_config {
+        ALLOWED_COBRANDS => [ { northamptonshire => '.' } ],
```
Comment (dracos) [inline t/cobrand/northamptonshire.t:267]: This can be `ALLOWED_COBRANDS => 'northamptonshire'`

**Review by dracos — CHANGES_REQUESTED** [review]
Couple of small comments

File: t/cobrand/northamptonshire.t (hunk 2 above)
Comment (struan) [reply t/cobrand/northamptonshire.t:270]: It's to enable us to display a status on the front end based on a state and external status code. This is to handle things like the further investigation state that NCC want, to which they've added a few more.

Comment (dracos) [reply t/cobrand/northamptonshire.t:270]: Okay, but the code to do that isn't in this PR or on master? :)

Comment (dracos) [reply t/cobrand/northamptonshire.t:270]: Ignore me!

**Review by dracos — APPROVED** [review]
Woo. Sorry about that!

**Pull request merged**